legal-api/src/legal_api/services/filings/validations/amalgamation_application.py

```diff
@@ -98,9 +101,15 @@ def validate_amalgamating_businesses(  # pylint: disable=too-many-branches,too-m
                 is_any_ccc = True
             elif business.legal_type == Business.LegalTypes.BC_ULC_COMPANY.value:
                 is_any_ulc = True
-        elif ((corp_number := amalgamating_business_json.get('corpNumber')) and corp_number.startswith('A')):
-            if ((foreign_jurisdiction := amalgamating_business_json.get('foreignJurisdiction')) and
-                    foreign_jurisdiction.get('country') == 'CA' and foreign_jurisdiction.get('region') == 'BC'):
+        elif corp_number := amalgamating_business_json.get('corpNumber'):
+            if corp_number in amalgamating_businesses:
+                duplicate_businesses.append(corp_number)
+            amalgamating_businesses[corp_number] = amalgamating_business_json
```

Comment:
`amalgamating_businesses` list is used to hold `Business` object (ie. a businesses which are in lear db). if you include foreign business into it then the below code fails to (`if amalgamating_business:`) identify the business

Reply:
just realized we get the amalgamating_business using identifier below. it should work